cub/cub/warp/specializations/warp_exchange_smem.cuh

```diff
@@ -0,0 +1,181 @@
+/******************************************************************************
+ * Copyright (c) 2011-2023, NVIDIA CORPORATION.  All rights reserved.
```

Comment:
Ditto copyright

Reply:
this one is 100% not copyrightable under my name, since it's just a copypasta move from `warp_exchange.cuh`